### ISSUE TITLE
CTE  not working with newer version 21.3.14.1 and with 21.7  on multi-node cluster

### ISSUE THREAD

**Issue by DipalPrajapati** [ClickHouse/ClickHouse#26956]
Please bear the complex query , I do have 4 node clickhouse cluster.  
I recently migrated from 20.x to 21.3.14.1 . The existing queries are breaking. 
Version :  21.3.14.1

```
INSERT INTO db1.mytale
            (
                        my fields
            )
            WITH temp AS
            (
                     SELECT   x
                     FROM     db1.tablex final
                     WHERE    time < '2021-07-28 14:49:31'
                     GROUP BY x
            )SELECT *
   FROM   (
                     SELECT     myfields
                     FROM       (
                                         SELECT   x,
                                                  y,
                                                  other aggergation /anylast feilds
                                         FROM     db1.tablex final
                                         WHERE    isnotnull(x)
                                         AND      msisdn != '' x IN
                                                  (
                                                           SELECT   x
                                                           FROM     db1.tablex final
                                                           WHERE    time < '2021-07-28 14:49:31'
                                                           GROUP BY x
                                                           HAVING   count(DISTINCT y) > 1)
                                         GROUP BY x,
                                                  y,
                                                  
                                         HAVING   time < '2021-07-28 14:49:31' ) AS e1 global asof
                     INNER JOIN
                                (
                                         SELECT   x,
                                                  y,
                                                  other aggergation /anylast feilds
                                         FROM     db1.tablex final
                                         WHERE    isnotnull(x)
                                         AND      x IN
                                                  (
                                                         SELECT x
                                                         FROM   temp)
                                         GROUP BY x,
                                                  y
                                         HAVING   time < '2021-07-28 14:49:31' settings distributed_group_by_no_merge =1) AS e2
                     ON         (
                                           e1.x = e2.x)
                     AND        (
                                           e1.time < e2.time)) settings distributed_group_by_no_merge = 1

``` 
 DB::Exception: Received from clickhouse-distributed-0-1:9000. DB::Exception: Table default.temp doesn't exist.

This say default.temp does not exists. 

#20187 
As per this the given syntax works 

`create table t engine = Memory as
with cte as (select * from numbers(10))
select * from cte
`

Can you please help? The same was working with 20.x version. 
I tested the same with 21.7 which is latest.

**Comment by den-crane** [comment]
Yesterday 21.7.5.29 have been released. Did you try with 21.7.5.29?

It's working for me 

```
create table t engine = Memory as with cte as (select * from numbers(10)) select * from cte

Ok.

0 rows in set. Elapsed: 0.006 sec.

21.9.1.7603.
```

**Comment by den-crane** [comment]
Also there are new parameters try to disable them

```
  --allow_push_predicate_when_subquery_contains_with arg           Allows push predicate when subquery contains WITH clause
  --enable_global_with_statement arg                               Propagate WITH statements to UNION queries and all subqueries
```

**Comment by DipalPrajapati** [comment]
yes ,I tried with 21.7.5.29. 
allow_push_predicate_when_subquery_contains_with
enable_global_with_statement 

Do i need to disable them ? would you tell me the values? 0 ?

**Comment by DipalPrajapati** [comment]
> Yesterday 21.7.5.29 have been released. Did you try with 21.7.5.29?
> 
> It's working for me
> 
> ```
> create table t engine = Memory as with cte as (select * from numbers(10)) select * from cte
> 
> Ok.
> 
> 0 rows in set. Elapsed: 0.006 sec.
> 
> 21.9.1.7603.
> ```

but  on cluster too?

**Comment by den-crane** [comment]
>but on cluster too?

What do you mean? I don't understand your question.

**Comment by den-crane** [comment]
> yes ,I tried with 21.7.5.29.
> allow_push_predicate_when_subquery_contains_with
> enable_global_with_statement
> 
> Do i need to disable them ? would you tell me the values? 0 ?

yes, set it zero in a user profile

**Comment by DipalPrajapati** [comment]
allow_push_predicate_when_subquery_contains_with
enable_global_with_statement

Would disabling above settings impact the performance of the query ?

**Comment by DipalPrajapati** [comment]
```

SELECT *
FROM system.settings
WHERE name = 'enable_global_with_statement'

Query id: 02920909-739c-48d9-9bb5-c2c9bc20aec9

┌─name─────────────────────────┬─value─┬─changed─┬─description───────────────────────────────────────────────────┬─min──┬─max──┬─readonly─┬─type─┐
│ enable_global_with_statement │ 0     │       1 │ Propagate WITH statements to UNION queries and all subqueries │ ᴺᵁᴸᴸ │ ᴺᵁᴸᴸ │        0 │ Bool │
└──────────────────────────────┴───────┴─────────┴───────────────────────────────────────────────────────────────┴──────┴──────┴──────────┴──────┘

1 rows in set. Elapsed: 0.006 sec.

 select * from system.settings where name='allow_push_predicate_when_subquery_contains_with';

SELECT *
FROM system.settings
WHERE name = 'allow_push_predicate_when_subquery_contains_with'

Query id: bc279f33-d37f-4236-89f3-7a39590c5d4b

┌─name─────────────────────────────────────────────┬─value─┬─changed─┬─description──────────────────────────────────────────────┬─min──┬─max──┬─readonly─┬─type─┐
│ allow_push_predicate_when_subquery_contains_with │ 0     │       1 │ Allows push predicate when subquery contains WITH clause │ ᴺᵁᴸᴸ │ ᴺᵁᴸᴸ │        0 │ Bool │
└──────────────────────────────────────────────────┴───────┴─────────┴──────────────────────────────────────────────────────────┴──────┴──────┴──────────┴──────┘

1 rows in set. Elapsed: 0.009 sec.

```
Do not see it working !

**Comment by DipalPrajapati** [comment]
Even below is not working.
Though I do not need that functionality , but for your information , as I see it is resolved in  #22657
```
create view v as
with cte as (select * from numbers(10))
select * from cte
```
